tasks/main.yml

```diff
@@ -1 +1,155 @@
-# role tasks
+# Tasks are insipred and named after the Red Hat OpenStack install directions
+- name: 4.2 Creating directories for templates and images
+  become: "{{ osp_director_become }}"
+  become_user: "{{ osp_director_become_user }}"
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0755
+  with_items:
+    - ~/images
```

Comment:
That seems like a great theory, but filters are interpreted locally not remotely, yeah? In which case I would get "/home/ghelling/images" here and not "/home/stack/images". I submit the following as evidence - on my local machine my home is "/home/ghelling" while on 1chronicles machine my local directory should be "/home/greg".

```shell
(tools3) [ghelling@work 🎩 osp_director] (initial)$ ansible -i ~/SpiderOak\ Hive/inventory/private/hosts -m debug -a 'msg={{ "~/" | expanduser }}' all,localhost
1chronicles | SUCCESS => {
    "msg": "/home/ghelling/"
}
localhost | SUCCESS => {
    "msg": "/home/ghelling/"
}
```

Reply:
Yeah, that's definitely true, so I'll quit it with the expanduser advice, or at least not forget about the caveat of where filters run. I've run into modules that don't expand `~`, there's no easy way to reliably get a reference to the absolute path that `~` actually is for inserting into things like config files (like with template), and the ambiguity of what `~` is when `become`ing are good reasons, for me, to try to avoid using `~` entirely in generic roles on the remote side.

If the remote homedir for a specific user is really needed, then probably going through the whole process of `getent passwd <user> | cut -d: -f6` or some similar thing to get the homedir absolute path should be done.

That all said...I don't know anything about the images dir, and if `~/images/` works then...cool!

Reply:
As long as you're not "become"-ing, then `ansible_user_dir` works great.

If you're using "become" or dropping something into the home directory for a user who is not the login user or root, then you're pretty well screwed. Maybe we can propose adding user info as an enhancement to the setup module in Ansible if we really need these things in the future.

As an aside, I would suggest that any module which does not understand "~/" needs to have an enhancement/bug filed against it. It's a simple Python call away, so expanding that should not be a difficult thing for the module to do. Even the `expanduser` filter is just a simple wrapping of the built-in Python function with logic to massage its output to a proper string based on Python 2/3 variability.

Reply:
I think the takeaway here is that I'm gonna mark all those links to my expanduser comment as resolved. :)